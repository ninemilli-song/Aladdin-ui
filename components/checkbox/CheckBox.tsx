import React from 'react';
import PropTypes from 'prop-types';

import { CheckBoxProps, KindMap } from './interface';

const kinds: KindMap = {
  info: '#5352ED',
  positive: '#2ED573',
  negative: '#FF4757',
  warning: '#FFA502',
};

const prefixCls = 'aladdin-checkbox';

const CheckBox: React.FC<CheckBoxProps> = ({ children, kind = 'info', ...rest }) => {
  return (
    <div
      className={prefixCls}
      style={{
        background: kinds[kind],
      }}
      {...rest}
    >
      {children}
    </div>
  );
};

CheckBox.propTypes = {
  kind: PropTypes.oneOf(['info', 'positive', 'negative', 'warning']),
};

CheckBox.defaultProps = {
  kind: 'info',
};

export default CheckBox;
