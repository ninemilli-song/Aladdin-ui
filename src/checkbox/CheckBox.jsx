import React from 'react';
import './style/index.less';

const CheckBox = (props) => {
    return (
        <div className='aladdin-checkbox'>
            Hello CheckBox!
        </div>
    )
};

export default CheckBox;
