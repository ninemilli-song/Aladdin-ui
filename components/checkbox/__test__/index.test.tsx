import React from 'react';
import { render } from '@testing-library/react';
import CheckBox from '../CheckBox';

describe('<CheckBox />', () => {
  test('should render default', () => {
    const { container } = render(<CheckBox>default</CheckBox>);
    expect(container).toMatchSnapshot();
  });

  test('should render checkbox with type', () => {
    const kinds: any[] = ['info', 'warning', 'positive', 'negative'];

    const { getByText } = render(
      <>
        {kinds.map((k) => (
          <CheckBox kind={k} key={k}>
            {k}
          </CheckBox>
        ))}
      </>,
    );

    kinds.forEach((k) => {
      expect(getByText(k)).toMatchSnapshot();
    });
  });
});
