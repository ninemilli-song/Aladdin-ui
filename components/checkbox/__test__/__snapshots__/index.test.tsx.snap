// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> should render checkbox with type 1`] = `
<div
  class="aladdin-checkbox"
  style="background: rgb(83, 82, 237);"
>
  info
</div>
`;

exports[`<CheckBox /> should render checkbox with type 2`] = `
<div
  class="aladdin-checkbox"
  style="background: rgb(255, 165, 2);"
>
  warning
</div>
`;

exports[`<CheckBox /> should render checkbox with type 3`] = `
<div
  class="aladdin-checkbox"
  style="background: rgb(46, 213, 115);"
>
  positive
</div>
`;

exports[`<CheckBox /> should render checkbox with type 4`] = `
<div
  class="aladdin-checkbox"
  style="background: rgb(255, 71, 87);"
>
  negative
</div>
`;

exports[`<CheckBox /> should render default 1`] = `
<div>
  <div
    class="aladdin-checkbox"
    style="background: rgb(83, 82, 237);"
  >
    default
  </div>
</div>
`;
